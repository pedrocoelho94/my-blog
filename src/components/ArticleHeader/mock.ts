import { data } from 'api/mockPosts.json'

const { id, title, excerpt, cover, author, categories, createdAt } =
  data.posts[0]

export default {
  id,
  title,
  excerpt,
  cover,
  author,
  categories,
  createdAt
}
