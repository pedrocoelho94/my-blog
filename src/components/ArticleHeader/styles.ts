import styled, { css } from 'styled-components'
import * as HeadingStyles from 'components/Heading/styles'

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles.Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      font-size: ${theme.font.sizes.large};
    }
  `}
`

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: calc(${theme.font.sizes.medium} * 0.9);
  `}
`

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`
