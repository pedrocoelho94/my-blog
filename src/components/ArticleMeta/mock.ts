export default {
  createdAt: '2021-03-02T15:16:09.246+00:00',
  author: {
    id: '123456789',
    slug: 'pedro-coelho',
    displayName: 'Pedro Coelho'
  },
  categories: [
    {
      id: '78724ec23f',
      displayName: 'Tech',
      slug: 'tech'
    },
    {
      id: '762vg214g34',
      displayName: 'JS',
      slug: 'javascript'
    }
  ]
}
