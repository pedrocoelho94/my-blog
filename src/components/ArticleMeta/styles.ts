import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child):after {
      content: ', ';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 0.2s ease-in-out;

      &:hover {
        filter: brightness(0.5);
      }
    }
  `}
`
