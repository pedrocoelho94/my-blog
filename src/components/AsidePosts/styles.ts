import styled, { css } from 'styled-components'

export const Wrapper = styled.div``

export const Content = styled.a`
  ${({ theme }) => css`
    display: flex;
    gap: 1rem;
    height: 100%;
    cursor: pointer;
    text-decoration: none;
    margin: ${theme.spacings.medium} 0;
  `}
`

export const Cover = styled.img`
  border-radius: 0.8rem;
  width: 10rem;
  height: 10rem;
  object-fit: cover;
`

export const Title = styled.h4`
  ${({ theme }) => css`
    font-size: 1.5rem;
    height: 100%;
    margin: 0;
    color: ${theme.colors.darker};

    &:hover {
      text-decoration: underline;
    }
  `}
`
