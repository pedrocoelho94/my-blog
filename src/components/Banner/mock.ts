import { data } from 'api/mockPosts.json'

export default {
  settings: data.setting,
  post: data.posts[0]
}
