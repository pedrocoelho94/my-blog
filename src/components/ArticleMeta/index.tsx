import { Author } from 'shared-typed/author'
import { Category } from 'shared-typed/category'
import { formatDate } from 'utils/format-date'
import * as S from './styles'

export type ArticleMetaProps = {
  createdAt: string
  author?: Author
  categories?: Category[]
}

const ArticleMeta = ({
  author = undefined,
  categories = [],
  createdAt
}: ArticleMetaProps) => (
  <S.Wrapper>
    <p>
      {typeof author !== 'undefined' && (
        <>
          <span>Por </span>
          <a href={`/author/${author.slug}`}>{author.displayName}</a>
          <span className="separator"> em </span>
        </>
      )}

      <time dateTime={createdAt}>{formatDate(createdAt)}</time>

      {categories.length > 0 && (
        <>
          <span className="separator"> - </span>
          <span className="categories">
            {categories.map((category) => (
              <span key={`article-meta-cat-${category.id}`}>
                <a href={`/category/${category.slug}`}>
                  {category.displayName}
                </a>
              </span>
            ))}
          </span>
        </>
      )}
    </p>
  </S.Wrapper>
)

export default ArticleMeta
