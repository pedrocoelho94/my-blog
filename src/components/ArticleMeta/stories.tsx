import { Story, Meta } from '@storybook/react'
import ArticleMeta, { ArticleMetaProps } from '.'

import mock from './mock'

export default {
  title: 'Posts/ArticleMeta',
  component: ArticleMeta,
  args: mock,
  argTypes: {
    createdAt: {
      control: {
        type: 'date'
      }
    },
    author: { type: null },
    categories: { type: null }
  }
} as Meta

export const Default: Story<ArticleMetaProps> = (args) => (
  <div>
    <ArticleMeta {...args} />
  </div>
)
