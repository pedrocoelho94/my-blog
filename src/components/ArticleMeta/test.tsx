import { screen } from '@testing-library/react'
import { renderWithTheme } from 'styles/renderWithTheme'

import ArticleMeta, { ArticleMetaProps } from '.'
import mock from './mock'

const props: ArticleMetaProps = mock

describe('<ArticleMeta />', () => {
  it('should render author and category links', () => {
    renderWithTheme(<ArticleMeta {...props} />)

    expect(
      screen.getByRole('link', { name: props.author.displayName })
    ).toHaveAttribute('href', `/author/${props.author.slug}`)

    expect(screen.getByRole('link', { name: /tech/i })).toHaveAttribute(
      'href',
      `/category/tech`
    )

    expect(screen.getByRole('link', { name: /js/i })).toHaveAttribute(
      'href',
      `/category/javascript`
    )
  })

  it('should format date', () => {
    const { container } = renderWithTheme(<ArticleMeta {...props} />)

    expect(screen.getByText('2 de mar. de 2021')).toHaveAttribute(
      'datetime',
      props.createdAt
    )

    expect(container).toMatchSnapshot()
  })
})
