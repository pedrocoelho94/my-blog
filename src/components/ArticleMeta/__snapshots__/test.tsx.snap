// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleMeta /> should format date 1`] = `
.c0 {
  font-size: 1.6rem;
  color: #AAAAAA;
  font-style: italic;
}

.c0 .categories span:not(:last-child):after {
  content: ', ';
}

.c0 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(0.5);
  filter: brightness(0.5);
}

<div>
  <div
    class="c0"
  >
    <p>
      <span>
        Por 
      </span>
      <a
        href="/author/pedro-coelho"
      >
        Pedro Coelho
      </a>
      <span
        class="separator"
      >
         em 
      </span>
      <time
        datetime="2021-03-02T15:16:09.246+00:00"
      >
        2 de mar. de 2021
      </time>
      <span
        class="separator"
      >
         - 
      </span>
      <span
        class="categories"
      >
        <span>
          <a
            href="/category/tech"
          >
            Tech
          </a>
        </span>
        <span>
          <a
            href="/category/javascript"
          >
            JS
          </a>
        </span>
      </span>
    </p>
  </div>
</div>
`

exports[
  `<ArticleMeta /> should match snapshot with no author and categories 1`
] = `
.c0 {
  font-size: 1.6rem;
  color: #AAAAAA;
  font-style: italic;
}

.c0 .categories span:not(:last-child):after {
  content: ', ';
}

.c0 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(0.5);
  filter: brightness(0.5);
}

<div>
  <div
    class="c0"
  >
    <p>
      <time
        datetime="2021-03-02T15:16:09.246+00:00"
      >
        2 de mar. de 2021
      </time>
    </p>
  </div>
</div>
`
